technologies/job/azure-machine-learning/automated-ml/instanceActions.js

```diff
@@ -21,15 +22,97 @@ exports.start = async ({ job, instance }) => {
+          'enable_onnx_compatible_models': false,
+          'max_cores_per_iteration': -1,
+          'send_telemetry': true,
+          'enable_distributed_featurization': true,
+          'compute_target': job.featuresValues.compute.label,
+          'enable_dnn': false,
+          'experiment_exit_score': null,
+          'experiment_timeout_minutes': 180,
+          'featurization': 'auto',
+          'is_timeseries': false,
+          'iteration_timeout_minutes': 180,
+          'max_concurrent_iterations': 6,
+          'metric_operation': 'maximize',
+          'model_explainability': true,
+          'n_cross_validations': null,
+          'name': job.featuresValues.experiment.label,
+          'path': `./sample_projects/${job.featuresValues.experiment.label}`,
+          'preprocess': true,
+          'primary_metric': 'accuracy',
+          'region': 'centralus',
+          'resource_group': job.featuresValues.resourceGroup.label,
+          'subscription_id': job.featuresValues.endpoint.subscriptionId,
+          'task_type': 'classification',
+          'validation_size': null,
+          'vm_type': 'STANDARD_D1',
+          'workspace_name': job.featuresValues.workspace.label,
+        };
+
+        const runProperties = {
+          numIterations: 1000,
+          metrics :["accuracy"],
+          primaryMetric :"accuracy",
+          trainSplit: 0,
+          acquisitionParameter: 0,
+          numCrossValidation: 5,
+          target: job.featuresValues.compute.label,
+          amlSettingsJsonString: JSON.stringify(amlSettings),
+          dataPrepJsonString: JSON.stringify(dataPrep),
+          enableSubsampling: false,
+          scenario: "UI"
+        };
+
+        const headersWithAccessToken = await getHeadersWithAccessTokenForManagementResource(job.featuresValues.endpoint);
+
+        const { data: runId } = await axios.post(
+          `${apiUrl}/jasmine/v1.0/subscriptions/${job.featuresValues.endpoint.subscriptionId}/resourceGroups/${job.featuresValues.resourceGroup.label}/providers/Microsoft.MachineLearningServices/workspaces/${job.featuresValues.workspace.label}/experiment/${job.featuresValues.experiment.label}/run`,
+          runProperties,
+          headersWithAccessToken,
+        );
+
+        const jsonDefinition = `{"Configuration":{"script":"train.py","arguments":[],"target":"${job.featuresValues.compute.label}","framework":"python","communicator":"None","autoPrepareEnvironment":true,"maxRunDurationSeconds":null,"nodeCount":1,"environment":null,"history":{"outputCollection":true,"snapshotProject":true,"directoriesToWatch":["logs"]},"spark":{"configuration":{"spark.app.name":"Azure ML Experiment","spark.yarn.maxAppAttempts":1}},"hdi":{"yarnDeployMode":"cluster"},"tensorflow":{"workerCount":1,"parameterServerCount":1},"mpi":{"processCountPerNode":1},"dataReferences":{},"sourceDirectoryDataStore":null,"amlcompute":{"vmSize":null,"vmPriority":null,"retainCluster":false,"name":null,"clusterMaxNodeCount":1}}}`;
```

Comment:
If it is a JSON definition, why not writing it in plain JavaScript Object and `JSON.stringify()` it? It will be more maintainable than a `string`.

Reply:
Yes I do that